reducer/src/main/java/com/graphicsfuzz/reducer/reductionopportunities/VariableDeclToExprReductionOpportunities.java

```diff
@@ -53,4 +55,28 @@ public VariableDeclToExprReductionOpportunities(TranslationUnit tu, ReducerConte
     finder.visit(tu);
     return finder.getOpportunities();
   }
+
+  @Override
+  public void visitDeclarationStmt(DeclarationStmt declarationStmt) {
+    super.visitDeclarationStmt(declarationStmt);
+    if (!context.reduceEverywhere()) {
+      // Replacing variable declaration with a new binary expression might have a side-effect,
+      // do not consider these reduction opportunities if we are not reducing everywhere.
+      return;
+    }
+    final List<VariableDeclInfo> declInfos =
+        declarationStmt.getVariablesDeclaration().getDeclInfos();
+    // We iterate backwards so that when applying reduction the new expression is inserted at the
+    // correct order with respect to its original order in the variable declaration info list.
+    for (int i = declInfos.size() - 1; i >= 0; i--) {
+      final VariableDeclInfo variableDeclInfo = declInfos.get(i);
+      if (variableDeclInfo.hasInitializer()) {
+        addOpportunity(new VariableDeclToExprReductionOpportunity(
+            variableDeclInfo,
```

Comment:
Because the current class extends ScopeTreeBuilder, you have access to the "currentBlock" method which gets the block that visitation is inside.  I suggest you get rid of parentMap and call this method to get the block.

Reply:
Nice! thank you for guiding.